[CK_BUILDER] Update the testing documentation

shumway opened the pull request:
Now that we have clear sets for smoke tests and regression test, we rearrange the CMakeLists.txt file to be organized and have description and instructional comments.

Move all the test targets that compile quickly into the smoke test suite.

Update the builder README.md to reflect this new test organization and functionality.

## Proposed changes

Please describe the motivation behind the pull request, whether it enables a new feature or fixes a bug. If there are associated pull requests or issues, please link them to the pull request.

shumway committed:
[CKBuilder] Update the testing documentation

Now that we have clear sets for smoke tests and regression test, we rearange the CMakeLists.txt file to be organized and have description and instructional comments.

Move all the test targets that compile quickly into the smoke test suite.

Update the builder README.md to reflect this new test organization and functionality.

spolifroni-amd commented on experimental/builder/README.md:
````diff
@@ -35,27 +35,41 @@ cmake
+```
+
+### Regression Tests (Integration Tests)
+Integration tests that compile actual GPU kernels to verify the builder generates valid, compilable code. These are more expensive (may take minutes) but catch real-world issues.
````
```suggestion
Integration tests that compile actual GPU kernels to verify that the builder generates valid, compilable code. These are more expensive than smoke tests (can take minutes to run) but catch real-world issues.
```

spolifroni-amd reviewed (changes requested):
A small suggestion re. wording of the regression tests.

shumway committed:
Update experimental/builder/README.md

Clarify integration tests description from review comment.

Co-authored-by: spolifroni-amd <Sandra.Polifroni@amd.com>

shumway commented:
> A small suggestion re. wording of the regression tests.

@spolifroni-amd 

This PR is from five days ago and is blocking other PRs. I had hoped to merge it this morning.  I'm now rerunning CI which will take another 4-8 hours, and it looks like the readthedocs check is broken now, too. Can you please approve this PR so I can get it moving, and if you know how to fix the readthedocs error that would be appreciated. Thank you.

shumway committed:
Correct README.md

The regression tests here still run very fast like smoke tests, but can take minutes or even tens of minutes to compile.  We test most of the builder functionality without compiling heavily-templated kernel code, but these regression tests do an expensive full build of the CK kernels.

spolifroni-amd commented:
> > A small suggestion re. wording of the regression tests.
> 
> @spolifroni-amd
> 
> This PR is from five days ago and is blocking other PRs. I had hoped to merge it this morning. I'm now rerunning CI which will take another 4-8 hours, and it looks like the readthedocs check is broken now, too. Can you please approve this PR so I can get it moving, and if you know how to fix the readthedocs error that would be appreciated. Thank you.

It's approved. The RTD check seems to have passed, but if you see a specific issue with it, let me know.
